### PR TITLE
Updates python version in pyproject.toml and removes requirements.txt

### PR DESCRIPTION
**Pull request by Rdbaker**
### Purpose

This change:
* updates the versions of the python requirement to align with PUDL, so that this module can be installed
* removes the unused `requirements.txt` file, since we have 3 other dependency management files in this repo (correct me if we want that, but it wasn't referenced in the README at all)

### What the code is doing

updating the python version in pyproject.toml

### Testing

I will test this change by trying to install it in Greta

### Where to look

n/a

### Usage Example/Visuals

n/a

### Review estimate

5 minutes

### Future work

none for this change

### Checklist
- [x] Update the documentation to reflect changes made in this PR
- [x] Format all updated python files using `black`
- [x] Clear outputs from all notebooks modified
- [x] Add docstrings and type hints to any new functions created

### CODE REVIEW

**Comment by Rdbaker** [comment]
tested that it was able to be installed in Greta

**Comment by grgmiller** [comment]
Right now the target branch is `main` - if we test integration using this branch and everything works, we can merge directly into main and do a `v0.3.2` release of OGE; otherwise, we probably want to first merge into `development` in case there are any other patch changes we need to make during the testing process for integration.

**Comment by grgmiller** [comment]
If we update the version, this also needs to be updated in the `CITATION.cff` file along with the release date.

**Comment by Rdbaker** [comment]
> If we update the version, this also needs to be updated in the `CITATION.cff` file along with the release date.

why are we using both CITATION.cff and pyproject.toml to specify the version?

**Comment by grgmiller** [comment]
> why are we using both CITATION.cff and pyproject.toml to specify the version?

I think they serve different purposes. the citation file is to help with the citability for academic purposes, so the version number is specified there as well. https://citation-file-format.github.io/

File: pyproject.toml [hunk 1]
```diff
@@ -33,7 +33,6 @@ dependencies = [
   "statsmodels",
   "coloredlogs",
   "s3fs",
-  "catalystcoop-pudl@git+https://github.com/singularity-energy/pudl.git@oge_release",
```
Comment (grgmiller) [inline pyproject.toml:36]: We cannot remove PUDL from the dependencies of OGE

**Review by grgmiller — CHANGES_REQUESTED** [review]
We

**Review by grgmiller — APPROVED** [review]
Looks good

**Pull request merged**